src/Simplify_Call.cpp

```diff
@@ -577,7 +470,143 @@ Expr Simplify::visit(const Call *op, ExprInfo *bounds) {
                                         {std::move(cond), std::move(result), std::move(message)},
                                         Internal::Call::PureIntrinsic);
         }
-    } else {
+    } else if (op->call_type == Call::PureExtern) {
+        // TODO: This could probably be simplified into a single map-lookup
+        // with a bit more cleverness; not sure if the reduced lookup time
+        // would pay for itself (in comparison with the possible lost code clarity).
+
+        // Handle all the PureExtern cases of float -> bool
+        {
+            static const std::unordered_map<std::string, std::function<bool(double)>>
+                pure_externs_f1b = {
+                    {"is_finite_f16", [](double a) { return std::isfinite(a); }},
```

Comment:
Can you construct a std::function<bool(double)> from std::isfinite directly without the wrapping lambda? (and similar below)

Reply:
Done. (Note: if approved, please don't merge yet -- I want to re-run a global reality check at Google to be sure there are no injections from this.)